src/main/java/harmonised/pmmo/mixin/CampfireMixin.java

```diff
@@ -16,12 +16,11 @@
 
 @Mixin(CampfireBlockEntity.class)
 public class CampfireMixin {
-    @Inject(at=@At(
-            value = "INVOKE",
-            target = "Lnet/minecraft/world/Containers;dropItemStack(Lnet/minecraft/world/level/Level;DDDLnet/minecraft/world/item/ItemStack;)V"),
-    method = "cookTick(Lnet/minecraft/world/level/Level;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/entity/CampfireBlockEntity;)V",
-    locals = LocalCapture.CAPTURE_FAILHARD)
-    private static void projectmmo$handleCampfireCook(Level pLevel, BlockPos pPos, BlockState pState, CampfireBlockEntity pBlockEntity, CallbackInfo ci, boolean flag, int i, ItemStack itemstack, Container container, ItemStack itemstack1) {
+    @Inject(at=@At(value = "INVOKE",
+                   target = "Lnet/minecraft/world/Containers;dropItemStack(Lnet/minecraft/world/level/Level;DDDLnet/minecraft/world/item/ItemStack;)V"),
+            method = "cookTick(Lnet/minecraft/world/level/Level;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/entity/CampfireBlockEntity;)V",
+            locals = LocalCapture.CAPTURE_FAILHARD)
+    private static void postFurnaceBurnEvent(Level pLevel, BlockPos pPos, BlockState pState, CampfireBlockEntity pBlockEntity, CallbackInfo ci, boolean flag, int i, ItemStack itemstack, int j, Container container, ItemStack itemstack1) {
```

Comment:
This is a problem that only exists in dev because mojang and Forge use different decompilers.  This results in the locals recompiling differently in the two environments.  If you were to build your PR and run it in a production environment, your version would crash because of this mixin error.  I have a permanently un-committed change to this mixin in my development environment so that the working code goes to production and the development version with the extra local is present for testing.

Reply:
Huh... that explains a lot! I was wondering how could it be possible that you are actively developing this branch and yet it crashes on startup. I even began suspecting some hidden differences in the build environment... I almost got it :D

How did you set up this dev-only change on your side? Are you using git for that or some other tool?

Reply:
I use github actions to publish the mod, so since the mixin change is only on my local machine, when i push to origin and the publish process kicks off, it doesn't have that change.    There have been a couple of times where i forgot that change was in my staged changes and accidentally committed it to production.  luckily i don't have that issue in 1.21.